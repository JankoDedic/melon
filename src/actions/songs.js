import uuid from 'uuid';

export const addSong = (song) => {
  const { title = '', artists = '' } = song;
  return {
    type: 'ADD_SONG',
    song: { id: uuid.v4(), title, artists },
  };
};

export const editSong = (id, updates) => ({
  type: 'EDIT_SONG',
  id,
  updates,
});
