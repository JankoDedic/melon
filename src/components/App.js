import React from 'react';
import { connect } from 'react-redux';

import AppRouter from '../routers/AppRouter';
import Player from './Player';

export const App = ({ isUserAuthenticated }) => (
  <div>
    <AppRouter />
    {isUserAuthenticated && <Player />}
  </div>
);

const mapStateToProps = (state) => ({
  isUserAuthenticated: !!state.auth.uid,
});

export default connect(mapStateToProps)(App);
