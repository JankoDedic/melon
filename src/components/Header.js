import React from 'react';

export default () => (
  <div className="header-container">
    <div className="header">
      <div className="header__logo">
        Melon
      </div>
      <button className="header__logout">Logout</button>
    </div>
  </div>
);
