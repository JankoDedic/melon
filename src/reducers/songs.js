const defaultSongs = [
  { title: 'This Is What You Came For', artists: 'Calvin Harris, Rihanna' },
  { title: 'Your Way', artists: 'Frank Sinatra' }
];

export default (songs = defaultSongs, action) => {
  return songs;
};

