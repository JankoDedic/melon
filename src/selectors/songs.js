export default (songs, filters) => {
  return songs.filter((song) => {
    const doesTitleMatch = song.title.toLowerCase().includes(filters.text.toLowerCase());
    const doArtistsMatch = song.artists.toLowerCase().includes(filters.text.toLowerCase());
    return doesTitleMatch || doArtistsMatch;
  });
};
