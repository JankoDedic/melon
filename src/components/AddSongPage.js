import React from 'react';
import { connect } from 'react-redux';

import SongForm from './SongForm';
import { startAddSong } from '../actions/songs';

class AddSongPage extends React.Component {
  handleSubmit = (song) => {
    this.props.startAddSong(song);
    this.props.history.push('/dashboard');
  }
  render() {
    return (
      <SongForm formTitle="Add Song" onSubmit={this.handleSubmit} />
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddSong: (song) => dispatch(startAddSong(song)),
});

export default connect(undefined, mapDispatchToProps)(AddSongPage);
