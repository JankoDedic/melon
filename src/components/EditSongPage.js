import React from 'react';
import { connect } from 'react-redux';

import SongForm from './SongForm';
import { startEditSong, removeSong } from '../actions/songs';

class EditSongPage extends React.Component {
  handleSubmit = (song) => {
    this.props.startEditSong(this.props.song.id, song);
    this.props.history.push('/dashboard');
  }
  handleRemove = () => {
    this.props.removeSong(this.props.song.id);
    this.props.history.push('/dashboard');
  }
  render() {
    return (
      <div>
        <SongForm
          formTitle="Edit Song"
          onSubmit={this.handleSubmit}
          song={this.props.song}
        />
        <button
          className="form-button"
          onClick={this.handleRemove}
        >
          Remove Song
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  song: state.songs.find((song) => song.id === props.match.params.id),
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  startEditSong: (id, updates) => dispatch(startEditSong(id, updates)),
  removeSong: (id) => dispatch(removeSong(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(EditSongPage);
