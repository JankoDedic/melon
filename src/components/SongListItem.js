import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { setNowPlayingSong } from '../actions/nowPlayingSong';

export const SongListItem = ({ song, setNowPlayingSong }) => (
  <div className="song-list-item">
    <div className="song-list-item__info" onClick={setNowPlayingSong}>
      <div className="song-list-item__info__title">{song.title}</div>
      <div className="song-list-item__info__artists">{song.artists}</div>
    </div>
    <Link className="router-link song-list-item__edit_button" to={`/edit/${song.id}`}>edit</Link>
  </div>
);

const mapDispatchToProps = (dispatch, ownProps) => ({
  setNowPlayingSong: (song) => dispatch(setNowPlayingSong(ownProps.song)),
});

export default connect(undefined, mapDispatchToProps)(SongListItem);
