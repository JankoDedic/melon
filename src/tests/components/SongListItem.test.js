import React from 'react';
import { shallow } from 'enzyme';

import { SongListItem } from '../../components/SongListItem';
import songs from '../fixtures/songs';

test('SongListItem snapshot', () => {
  const wrapper = shallow(<SongListItem song={songs[0]} />);
  expect(wrapper).toMatchSnapshot();
});

describe('info area click', () => {
  const setNowPlayingSong = jest.fn();
  const song = songs[0];
  const wrapper = shallow(
    <SongListItem
      song={song}
      setNowPlayingSong={setNowPlayingSong}
    />
  );

  wrapper.find('.song-list-item__info').simulate('click');

  it('calls setNowPlayingSong', () => {
    expect(setNowPlayingSong).toHaveBeenCalled();
  });
});
