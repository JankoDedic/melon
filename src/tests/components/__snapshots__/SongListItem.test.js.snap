// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongListItem snapshot 1`] = `
<div
  className="song-list-item"
>
  <div
    className="song-list-item__info"
  >
    <div
      className="song-list-item__info__title"
    >
      Feeling Good
    </div>
    <div
      className="song-list-item__info__artists"
    >
      Muse
    </div>
  </div>
  <Link
    className="router-link song-list-item__edit_button"
    replace={false}
    to="/edit/1"
  >
    edit
  </Link>
</div>
`;
